validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/GetGasLimit.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright ConsenSys Software Inc., 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.restapi.apis;
+
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_FOUND;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
+import static tech.pegasys.teku.validator.client.restapi.ValidatorRestApi.TAG_GAS_LIMIT;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.tuweni.bytes.Bytes48;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.infrastructure.json.types.CoreTypes;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.json.types.StringBasedPrimitiveTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.ParameterMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.validator.client.BeaconProposerPreparer;
+import tech.pegasys.teku.validator.client.restapi.ValidatorTypes;
+
+public class GetGasLimit extends RestApiEndpoint {
+
+  public static final String ROUTE = "/eth/v1/validator/{pubkey}/gas_limit";
+  private final Optional<BeaconProposerPreparer> beaconProposerPreparer;
+
+  private static final String PARAM_PUBKEY = "pubkey";
+
+  public static final ParameterMetadata<BLSPublicKey> PARAM_PUBKEY_TYPE =
```

Comment:
this pubkey should be used a lot, are we not sharing it? it'll be the same definition every time... if it's not shared, we should fix that.

Reply:
I added it to `ValidatorTypes`